File: MAP/Lab 2/ToyInterpreter/src/model/expressions/ConstExp.java
package model.expressions;

import model.adts.MyIDictionary;

public class ConstExp implements IExp {
    private int value;

    public ConstExp(int value) {
        this.value = value;
    }

    @Override
    public int eval(MyIDictionary<String, Integer> d) {
        return this.value;
    }

    @Override
    public String toString() {
        return " " + value;
    }
}

File: MAP/Lab10/ConsoleApp1/ConsoleApp1/Model/expressions/ConstExp.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ToyInterpreter.Model.expressions
{
    class ConstExp : IExp
    {
        private int value;

        public ConstExp(int value)
        {
            this.value = value;
        }

        public int eval(Dictionary<String, Int64> d)
        {
            return this.value;
        }
  
        public override String ToString()
        {
            return " " + value;
        }
    }
}
